app/src/main/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagViewModel.kt

```diff
@@ -0,0 +1,116 @@
+        is AsyncResult.Success -> _featureFlagList.value = processFeatureFlagList(result.value)
+        else -> _featureFlagList.value = listOf()
+      }
+    }
+  }
+
+  private fun processFeatureFlagList(ephemeralFeatureFlags: List<EphemeralFeatureFlag>):
+    List<FeatureFlagItemViewModel> {
+
+      return ephemeralFeatureFlags.map { ephemeralFeatureFlag ->
+        val featureFlagName = getFeatureFlagDisplayName(ephemeralFeatureFlag.id)
+        val syncStatusText = getSyncStatusText(ephemeralFeatureFlag.syncStatus)
+        val isResetAvailable = canReset(ephemeralFeatureFlag)
+
+        FeatureFlagItemViewModel(
+          featureFlagName = featureFlagName,
+          syncStatus = syncStatusText,
+          isResetAvailable = isResetAvailable,
+          currentValue = ephemeralFeatureFlag.currentValue,
+          syncStatusBackground = getSyncStatusBackground(syncStatusText)
+        )
+      }
+    }
+
+  private fun getFeatureFlagDisplayName(id: FeatureFlagId): String {
+    return when (id) {
+      FeatureFlagId.DOWNLOADS_SUPPORT -> "Downloads Support"
+      FeatureFlagId.EXTRA_TOPIC_TABS_UI -> "Extra Topic Tabs UI"
+      FeatureFlagId.LEARNER_STUDY_ANALYTICS -> "Learner Study Analytics"
+      FeatureFlagId.FAST_LANGUAGE_SWITCHING_IN_LESSON -> "Fast Language Switching In Lesson"
+      FeatureFlagId.LOGGING_LEARNER_STUDY_IDS -> "Logging Learner Study IDs"
+      FeatureFlagId.EDIT_ACCOUNTS_OPTIONS_UI -> "Edit Accounts Options UI"
+      FeatureFlagId.PERFORMANCE_METRICS_COLLECTION -> "Performance Metrics Collection"
+      FeatureFlagId.SPOTLIGHT_UI -> "Spotlight UI"
+      FeatureFlagId.INTERACTION_CONFIG_CHANGE_STATE_RETENTION ->
+        "Interaction Config Change State Retention"
+      FeatureFlagId.APP_AND_OS_DEPRECATION -> "App And OS Deprecation"
+      FeatureFlagId.NPS_SURVEY -> "NPS Survey"
+      FeatureFlagId.ONBOARDING_FLOW_V2 -> "Onboarding Flow V2"
+      FeatureFlagId.MULTIPLE_CLASSROOMS -> "Multiple Classrooms"
+      FeatureFlagId.FLASHBACK_SUPPORT -> "Flashback Support"
+      FeatureFlagId.UNRECOGNIZED, FeatureFlagId.FEATURE_FLAG_ID_UNSPECIFIED -> "Unknown Feature"
+    }
+  }
+  private fun getSyncStatusText(syncStatus: SyncStatus): String {
+    return when (syncStatus) {
+      SyncStatus.SYNC_STATUS_UNSPECIFIED -> "Unknown"
+      SyncStatus.NOT_SYNCED_FROM_SERVER -> "Default"
+      SyncStatus.SYNCED_FROM_SERVER -> "Server"
+      else -> "Unknown"
+    }
+  }
+
+  /**
+   * Returns the background drawable resource ID corresponding to the provided sync status string.
+   *
+   * @param syncStatus the string representation of the sync status for which to retrieve the background.
+   * @return the drawable resource ID to use as the background for this sync status.
+   */
+  fun getSyncStatusBackground(syncStatus: String): Int {
+    return when (syncStatus) {
+      "Server" -> R.drawable.rounded_rec_large_border_radius_green
+      "Overriden" -> R.drawable.rounded_rect_large_border_radius_yellow
```

Comment:
Here and elsewhere (if mentioned)
```suggestion
      "Overridden" -> R.drawable.rounded_rect_large_border_radius_yellow
```

Reply:
this is not under the scope of this PR so removed this